### PR TITLE
add dumping stack traces to chrome tracing file

### PR DESCRIPTION
**Pull request by gaoteng-git**
In order to show stack-traces in tensorboard plugin, we have to dump stack traces to chrome tracing file.
If this is merged, we will submit PR to torch/csrc/autograd/profiler_kineto.cpp, mainly adding following code into finalizeCPUTrace:
      if (kineto_events_[idx].hasStack()) {
        cpu_trace->activities[idx].callStack = stacksToStr(kineto_events_[idx].stack());
      }

### CODE REVIEW

File: libkineto/include/ClientTraceActivity.h [hunk 1]
```diff
@@ -69,6 +69,7 @@ struct ClientTraceActivity : TraceActivity {
   std::string outputTypes;
   std::string inputNames;
   std::string outputNames;
+  std::string stackTraces;
```
Comment (gdankel) [inline libkineto/include/ClientTraceActivity.h:72]: Can we use callStack and Call stack?

**Comment by gdankel** [comment]
Did you consider using the "sf" or "stack" fields for this? Also, can you point to some context - I might've missed parts of the discussion around this.

**Comment by gaoteng-git** [comment]
> Did you consider using the "sf" or "stack" fields for this? Also, can you point to some context - I might've missed parts of the discussion around this.

Yes, we considered this before. You can find reference [here](https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview#heading=h.yr703knxre9f). From my try, "sf" works but ["stack" seems not implemented](https://github.com/catapult-project/catapult/issues/3996).
Because one advantage of "sf" is decreasing file size. We compared the chrome tracing file size. Profiling the the resnet50 tranining for 6 steps, the chrome tracing file without dumping callstacks is 26MB, the chrome tracing file with dumping callstacks as this PR is 40MB. So the increased size is not too much.
Another advantage of "sf" is showing call stacks in multiple lines in chrome, like this:
image

if using "sf", the estimated change is changing the "stackTraces" in "ClientTraceActivity" from string to vector<string>(each element is a frame of call stack), and in kineto side adding a function to organize all operators' stack frames into a tree, and dumping the tree into a global secion "stackFrames" in the result json file.

What's your opinion on this? Should we adopt "sf" or keep this in this version?

File: libkineto/include/ClientTraceActivity.h (hunk 1 above)
Comment (gaoteng-git) [reply libkineto/include/ClientTraceActivity.h]: Of course, updated.

Comment (gaoteng-git) [reply libkineto/include/ClientTraceActivity.h]: > Can we use callStack and Call stack?

Of course, updated.

**Comment by gdankel** [comment]
Thanks I recall the previous discussion now. I'm fine with landing this now and updating it in a future version. Only potential issue will be to change the type later without breaking anything.

**Comment by gaoteng-git** [comment]
> Thanks I recall the previous discussion now. I'm fine with landing this now and updating it in a future version. Only potential issue will be to change the type later without breaking anything.

Thanks. Change which "type"? What do I need to do on my side?

**Review by gdankel — APPROVED** [review]

**Pull request closed without merging**